clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -363,12 +543,21 @@ public ConfigKey(String name, Type type, Object defaultValue, Validator validato
             if (this.validator != null)
                 this.validator.ensureValid(name, defaultValue);
             this.documentation = documentation;
+            this.dependents = dependents;
```

Comment:
Are parents/dependents ever validated? i.e. is there even a check that the dependent exists once all configs have been defined?